teuthology/scrape.py

```diff
@@ -410,7 +410,10 @@ def _get_service_types(self, job):
         result = defaultdict(list)
         # Lines like:
         # 2014-08-22T20:07:18.668 ERROR:tasks.ceph:saw valgrind issue   <kind>Leak_DefinitelyLost</kind> in /var/log/ceph/valgrind/osd.3.log.gz
-        for line in grep(os.path.join(job.path, "teuthology.log"), "</kind> in "):
+        valgrind_err_line = grep(os.path.join(job.path, "teuthology.log"), "</kind> in ")
+        valgrind_err_line = list(filter(None, valgrind_err_line)) # remove empty strings
```

Comment:
seems like the line below covers these cases. no?

Reply:
Okay so I wanted to get rid of both empty string, empty space and None. I just revised should be a one liner and covers all cases